server-spi/src/main/java/org/keycloak/services/clientpolicy/ClientPolicyEvent.java

```diff
@@ -32,6 +32,7 @@
     UNREGISTER,
     AUTHORIZATION_REQUEST,
     TOKEN_REQUEST,
+    SERVICE_ACCOUNT_TOKEN_REQUEST,
```

Comment:
You probably have a reason for a separate event but in theory, we could also leverage the `TOKEN_REQUEST` event but have some flag indicating whether the token request is coming from a client acting on its own behalf to obtain tokens.

I'm good to both approaches.

Reply:
Yes, exactly. That was exactly the approach I've started with. But in the end, I've used separate event as ServiceAccount has some small differences (Especially it does not have "parseResult" field, which is available only for the authorization_code grant in the TokenRequestContext, so for serviceAccount it would be nuill or more changes would be needed in the code of various executors etc... Due this I've used separate even in the end, which is also compatible with the approach by @tnorimat used for CIBA (BackchannelTokenRequestContext).